In [4]:
import pandas as pd
import glob as glob
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [5]:
def return_zoonomia_df(ENSTs):
    df = []
    for ENST in ENSTs:
        df.append(pd.read_csv("../soto_analysis/outputs/mutations/cds_zoonomia_all_TF_cds/" + ENST + ".bed", sep = "\t", header = None))
    return pd.concat(df)

In [6]:
# Loading in ALL TFs as the background
all_TFs = glob.glob("../soto_analysis/outputs/mutations/cds_zoonomia_all_TF_cds/ENST*")
all_TFs = set([_.split("/")[-1].split(".")[0] for _ in all_TFs])
len(all_TFs)

1558

In [7]:
zoonomia_cds = return_zoonomia_df(all_TFs)
zoonomia_cds

0          1          2                3  4   5          6          7  \
0     17   48613202   48613203  ENST00000239144 -1  17   48613202   48613203   
1     17   48613203   48613204  ENST00000239144 -1  17   48613203   48613204   
2     17   48613204   48613205  ENST00000239144 -1  17   48613204   48613205   
3     17   48613205   48613206  ENST00000239144 -1  17   48613205   48613206   
4     17   48613206   48613207  ENST00000239144 -1  17   48613206   48613207   
...   ..        ...        ...              ... ..  ..        ...        ...   
1687   5  179033636  179033637  ENST00000444149  1   5  179033636  179033637   
1688   5  179033637  179033638  ENST00000444149  1   5  179033637  179033638   
1689   5  179033638  179033639  ENST00000444149  1   5  179033638  179033639   
1690   5  179033639  179033640  ENST00000444149  1   5  179033639  179033640   
1691   5  179033640  179033641  ENST00000444149  1   5  179033640  179033641   

          8  
0     5.523  
1     5.501  
2     3.267  
3     5.523  
4     5.523  
...     ...  
1687 -0.084  
1688  1.596  
1689  5.608  
1690  2.786  
1691  3.933  

[2753376 rows x 9 columns]

In [77]:
pd.DataFrame(zoonomia_cds[8].value_counts())

count
8             
 6.335   43517
 6.360   42189
 8.796   39590
 8.903   38056
 6.332   11731
...        ...
-11.783      1
-12.108      1
-11.261      1
-13.033      1
-12.255      1

[21446 rows x 1 columns]

In [52]:
zoonomia_cds[zoonomia_cds[8] == 6.335].iloc[[0]]

0        1        2                3  4  5        6        7      8
1  Y  3579246  3579247  ENST00000321217  1  Y  3579246  3579247  6.335

In [51]:
zoonomia_cds[zoonomia_cds[8] == 6.360].iloc[[0]]

0         1         2                3  4   5         6         7     8
2566  19  56816244  56816245  ENST00000599577 -1  19  56816244  56816245  6.36

In [50]:
zoonomia_cds[zoonomia_cds[8] == 8.903].iloc[[0]]

0         1         2                3  4   5         6         7  \
2562  19  56816240  56816241  ENST00000599577 -1  19  56816240  56816241   

          8  
2562  8.903

In [53]:
zoonomia_cds[zoonomia_cds[8] == 8.796].iloc[[0]]

0         1         2                3  4   5         6         7  \
2544  19  56816222  56816223  ENST00000599577 -1  19  56816222  56816223   

          8  
2544  8.796

Download alignment at regions

chrY 3579246 3579247

chr19	56816244	56816245	

chr19	56816240	56816241	

chr19 56816222 56816223

In [66]:
start_to_PhyloP = {3579246: 6.335, 
                   56816244 : 6.36,
                   56816240 : 8.903,
                   56816222 : 8.796}

In [71]:
alignment = pd.read_csv("../data/zoonomia_test", sep = "\s+", skiprows = 1).reset_index()
alignment = alignment.rename(columns = {"score=1702000.000000" : "base"})
alignment

level_0                                            level_1     level_2  \
0         s                                  Homo_sapiens.chrY   3579246.0   
1         s                    Acinonyx_jubatus.LLWD01000177.1    644678.0   
2         s              Ailuropoda_melanoleuca.LNAT01000488.1  10197527.0   
3         s                     Ailurus_fulgens.LNAC01001076.1      7307.0   
4         s            Alouatta_palliata.AloPal_scaffold_61693      3214.0   
..      ...                                                ...         ...   
924       s         Vulpes_lagopus.VulLag_flattened_line_15794     39211.0   
925       s                Xerus_inauris.XerIna_scaffold_42695     13584.0   
926       s  Zalophus_californianus.ZalCal_flattened_line_1...     34944.0   
927       s               Zapus_hudsonius.ZapHud_scaffold_2224     72447.0   
928       s     Ziphius_cavirostris.ZipCav_flattened_line_1053     16292.0   

     level_3 level_4           a base  
0        1.0       +  57227415.0    G  
1        1.0       -    825115.0    G  
2        1.0       +  11087278.0    G  
3        1.0       -    380285.0    G  
4        1.0       -     21721.0    G  
..       ...     ...         ...  ...  
924      1.0       -     91214.0    G  
925      1.0       +     29930.0    G  
926      1.0       -     87988.0    G  
927      1.0       +    114907.0    G  
928      1.0       -     46223.0    G  

[929 rows x 7 columns]

In [72]:
alignment[alignment["a"].astype(str).str.contains("a")]

level_0               level_1  level_2  level_3 level_4   a base
185       a  score=2742285.000000      NaN      NaN     NaN NaN  NaN
433       a  score=3013500.000000      NaN      NaN     NaN NaN  NaN
681       a  score=3013500.000000      NaN      NaN     NaN NaN  NaN

In [73]:
alignment_1 = alignment.iloc[0:185]
alignment_2 = alignment.iloc[186:433]
alignment_3 = alignment.iloc[434:681]
alignment_4 = alignment.iloc[682:]

In [76]:
for alignment in [alignment_1, alignment_2, alignment_3, alignment_4]:
    #print(len(alignment))
    start = alignment[alignment["level_1"].str.contains("Homo_sapiens")]["level_2"].iloc[0]
    print("PhyloP " + str(start_to_PhyloP[start]))
    display(pd.DataFrame(alignment["base"].value_counts()))
    print()

PhyloP 6.335


count
base       
G       184
g         1


PhyloP 6.36


count
base       
T       246
-         1


PhyloP 8.903


count
base       
C       246
-         1


PhyloP 8.796


count
base       
G       245
g         1
-         1

In [43]:
alignment_1["score=1702000.000000"].value_counts()

score=1702000.000000
G    184
g      1
Name: count, dtype: int64

In [44]:
alignment_2 = alignment.iloc[186:]
alignment_2

level_0                                            level_1     level_2  \
186       s                                 Homo_sapiens.chr19  56816222.0   
187       s                    Acinonyx_jubatus.LLWD01001000.1    717743.0   
188       s             Acomys_cahirinus.AcoCah_scaffold_53323      7300.0   
189       s              Ailuropoda_melanoleuca.LNAT01000174.1    154940.0   
190       s                     Ailurus_fulgens.LNAC01000914.1    147050.0   
..      ...                                                ...         ...   
428       s         Vulpes_lagopus.VulLag_flattened_line_15794     39211.0   
429       s                Xerus_inauris.XerIna_scaffold_42695     13584.0   
430       s  Zalophus_californianus.ZalCal_flattened_line_1...     34944.0   
431       s               Zapus_hudsonius.ZapHud_scaffold_2224     72447.0   
432       s     Ziphius_cavirostris.ZipCav_flattened_line_1053     16292.0   

     level_3 level_4           a score=1702000.000000  
186      1.0       +  58617616.0                    G  
187      1.0       +   1065063.0                    G  
188      1.0       +     25475.0                    G  
189      1.0       -    865077.0                    G  
190      1.0       -    953532.0                    G  
..       ...     ...         ...                  ...  
428      1.0       -     91214.0                    G  
429      1.0       +     29930.0                    G  
430      1.0       -     87988.0                    G  
431      1.0       +    114907.0                    G  
432      1.0       -     46223.0                    G  

[247 rows x 7 columns]

In [45]:
alignment_2["score=1702000.000000"].value_counts()

score=1702000.000000
G    245
g      1
-      1
Name: count, dtype: int64

In [46]:
alignment_2[alignment_2["score=1702000.000000"] == "-"]

level_0                                level_1  level_2  level_3 level_4  \
282       s  Giraffa_tippelskirchi.LVKQ01S066458.1     39.0      0.0       -   

         a score=1702000.000000  
282  200.0                    -